pyccel/parser/syntax/openmp.py

```diff
@@ -390,15 +1027,380 @@ def expr(self):
+    def __init__(self, **kwargs):
+        """
+        """
+        self.tname = kwargs.pop('tname')
+        self.sname = kwargs.pop('sname', None)
+
+        super().__init__(**kwargs)
+
+    @property
+    def expr(self):
+        if DEBUG:
+            print("> Combined Taskloop Simd: expr")
+
+        txt = self.tname
+        if self.sname:
+            txt += ' ' + self.sname
+        return txt
+
+class OmpDistributeCombined(BasicStmt):
+    """Class representing a combined comstruct."""
+    def __init__(self, **kwargs):
+        """
+        """
+        self.dname  = kwargs.pop('dname')
+        self.sname  = kwargs.pop('sname', None)
+        self.pname  = kwargs.pop('pname', None)
+        self.fname  = kwargs.pop('fname', None)
+        self.ssname = kwargs.pop('ssname', None)
+
+        super().__init__(**kwargs)
+
+    @property
+    def expr(self):
+        if DEBUG:
+            print("> Combined Teams Distribute")
+
+        txt = self.dname
+        if self.sname:
+            txt += ' ' + self.sname
+        elif self.pname:
+            txt += ' ' + self.pname
+            txt += ' ' + self.fname
+            if self.ssname:
+                txt += ' ' + self.ssname
+
+        return txt
+
+class OmpTargetParallel(BasicStmt):
+    """Class representing a combined comstruct."""
+    def __init__(self, **kwargs):
+        """
+        """
+        self.pname = kwargs.pop('pname')
+        self.fname = kwargs.pop('fname', None)
+        self.sname = kwargs.pop('sname', None)
+
+        super().__init__(**kwargs)
+
+    @property
+    def expr(self):
+        if DEBUG:
+            print("> Combined Target Parallel")
+
+        txt = self.pname
+        if self.fname:
+            txt += ' ' + self.fname
+            if self.sname:
+                txt += ' ' + self.sname
+
+        return txt
+
+class OmpTargetTeams(BasicStmt):
+    """Class representing a combined comstruct."""
+    def __init__(self, **kwargs):
+        """
+        """
+        self.tname  = kwargs.pop('tname')
+        self.dname  = kwargs.pop('dname', None)
+        self.sname  = kwargs.pop('sname', None)
+        self.pname  = kwargs.pop('pname', None)
+        self.fname  = kwargs.pop('fname', None)
+        self.ssname = kwargs.pop('ssname', None)
+
+        super().__init__(**kwargs)
+
+    @property
+    def expr(self):
+        if DEBUG:
+            print("> Combined Target Teams")
+
+        txt = self.tname
+        if self.dname:
+            txt += ' ' + self.dname
+            if self.sname:
+                txt += ' ' + self.sname
+            else:
+                txt += ' ' + self.pname + ' ' + self.fname
+                if self.ssname:
+                    txt += ' ' + self.ssname
+        return txt
 #################################################
 
 #################################################
 # whenever a new rule is added in the grammar, we must update the following
 # lists.
+
+_valid_teams_clauses = (OmpPrivate,\
```

Comment:
By the way, I don't think the line continuation character `\` is necessary in a tuple.